ql/src/java/org/apache/hadoop/hive/ql/ddl/database/create/CreateDatabaseAnalyzer.java

```diff
@@ -78,10 +78,17 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
         ASTNode nextNode = (ASTNode) root.getChild(i);
         connectorName = ((ASTNode)nextNode).getChild(0).getText();
         outputs.add(toWriteEntity(connectorName));
-        if (managedLocationUri != null) {
-          outputs.remove(toWriteEntity(managedLocationUri));
-          managedLocationUri = null;
+
+        // HIVE-2436: Reject location and managed locations in DDL for REMOTE databases.
+        if (locationUri != null || managedLocationUri != null ) {
+          if (locationUri == null) {
+            outputs.remove(toWriteEntity(locationUri));
```

Comment:
so in HMS schema, "locationUri" is a required column. So I think HiveServer2 may have to send a location for a database, regardless of its type. The HiveMetastore might also use a default one. So a couple of things here
1) Seems like line 85 and 87 might result in a NullPointerException. Shouldn't we remove it when the value is NOT null?
2) given we are throwing an exception from this condition, do we even have to do outputs.remove() ? We are failing the operation anyway.

Reply:
Will change.